### DISCUSSION TITLE
Building cross-platform app

### DISCUSSION THREAD

**Discussion by AnkS4** [beeware/beeware#161] [General]
For building execuatable application for desktop platforms i.e. Linux, Mac & Windows, do I need to have all these systems? Can I use briefcase to make executables for all these platforms from Linux system?

**Comment by freakboy3742** [comment]
Unfortunately, the tools needed to package and sign macOS apps are only available for macOS; the tools needed to build Windows apps and MSI installers are only available for Windows; and the tools to build Linux AppImages are only available for Linux.

That said:
1. It is possible to build Linux AppImages on macOS using Docker. 
2. You don't need to have an operating system "on metal" to run those tools - it is possible to use an OS hosted in a virtual machine. 
3. It possible to use CI systems to build binaries. For example, Github Actions provides Linux, macOS and Windows host systems, and briefcase can run in all three environments. The only complication associated with this is getting signing credentials in place; but this is an otherwise fairly straightforward process of invoking briefcase in a CI configuration and uploading release binaries. Unfortunately, I can't point you at documentation for this process, but the Briefcase CI system builds some sample projects, which will hopefully be enough to get to started.
 
It may also be *possible* to use WINE to run the Windows tooling on Linux, but I haven't tried. I don't know any equivalent layer for macOS on Linux.
